packages/eslint-plugin/src/rules/no-misused-promises.ts

```diff
@@ -507,17 +510,41 @@ function voidFunctionParams(
       : subType.getConstructSignatures();
     for (const signature of signatures) {
       for (const [index, parameter] of signature.parameters.entries()) {
-        const type = checker.getTypeOfSymbolAtLocation(
+        const decl = parameter.getDeclarations()?.[0];
+        let type = checker.getTypeOfSymbolAtLocation(
           parameter,
           node.expression,
         );
+
+        const indices = [index];
+        // If this is a array 'rest' parameter, add all of the remaining parameter indices.
+        // Note - we currently do not support 'spread' arguments
+        if (
+          decl &&
+          ts.isParameter(decl) &&
+          decl.dotDotDotToken &&
+          checker.isArrayType(type) &&
+          node.arguments
```

Comment:
Similar note on removing this check not failing the tests now. In what situation would `node.arguments` not exist? Can we test for it?

Reply:
This occurs when invoking a constructor without providing parameters - e.g. `let a = new MyClass;`. I've added a single check at the top of the function, which just bails out since there's nothing to check.